docs/UserGuide.md

```diff
@@ -16,21 +23,85 @@ Cobb can also run on any machine, *without the need for an internet connection*.
+
+<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
+On first launch, Cobb will create a few files that have the extension `.json` in its *home directory*. These files are used
+by Cobb to store its data. **Edit these at your own risk**, as Cobb will start with an empty database if it detects any
+error in the formatting of the data in these files.
+</div>
+
+## Key definitions
+
+This section aims to define some terms that pop up throughout the user guide. These terms are used often, so it would be 
+good to take note of them before continuing.
+
+### Command-specific terms
+1. **Command**: A piece of instruction that a user types into the command input box to instruct Cobb's behaviour.
+   Commands have many variations, and can serve many purposes.
+2. **Syntax**: The manner in which is a command is to be typed. Can be thought of as the "format" of the command.
+3. **Parameter**: An input to a command. Can be thought of as a piece of information a user provides that determines the
+   behaviour of a specific command.
+4. **Command string**: The text that forms the command.
+5. **Flag**: A marker that signifies the start of a parameter to a command. For Cobb, in the command `findbuyer -k Tim`, 
+   `-k` is the flag, and `Tim` is the parameter.
+6. **Fuzzy**: A flag that signifies loose filtering, that is, given a set of filter requirements, any user that matches at least one
+   the given requirements will be returned.
+7. **Strict**: A flag that signifies tight filtering, that is, given a set of filter requirements, any user matching all of the
+   given requirements will be returned.
+8. **Ascending**: Related to order; from low to high or minimum to maximum.
+9. **Descending**: Related to order; from high to low or maximum to minimum. 
+
+### Person-specific terms
+1. **Name**: The name of the person.
+2. **Phone**: The phone number of the person.
+3. **Email**: The email of the person.
+4. **Address**: The address of the person.
+5. **Price Range**: The price range of properties that a person might consider buying. That is, any property whose price
+   falls within this range will be considered by the buyer.
+6. **Characteristics**: The characteristics of a property that a person desires. For example, a person that has characteristics
+   `bright; sunny` is ideally looking for a property that is also `bright` and `sunny`.
+7. **Priority**: The priority of the person. Can be `LOW`, `NORMAL` or `HIGH`.
+
+### Property-specific terms
+1. **Name**: The name of the property.
+2. **Price**: The price of the property.
+3. **Address**: The address of the property.
+4. **Description**: A short description of the property.
+5. **Characteristics**: The characteristics associated with a property. For example, a property that has characteristics
+   `windy; roomy` is both `windy` and `roomy`.
+6. **Owner**: The name of the owner of the property.
+7. **Phone**: The phone number of the owner of the property.
+
+
+
+## Layout
```

Comment:
I think Layout might be more important than talking about `property` and `buyer`, so maybe this should come before them